src/bin/makerd.rs

```diff
@@ -0,0 +1,88 @@
+use bitcoind::bitcoincore_rpc::Auth;
+use clap::Parser;
+use coinswap::{
+    maker::{start_maker_server, Maker, MakerBehavior},
+    utill::{
+        parse_proxy_auth, read_bitcoin_network_string, read_connection_network_string, setup_logger,
+    },
+    wallet::RPCConfig,
+};
+use std::{path::PathBuf, sync::Arc};
+
+/// The Maker Server.
+///
+/// This app starts the Maker server.
+#[derive(Parser)]
+#[clap(version = option_env ! ("CARGO_PKG_VERSION").unwrap_or("unknown"),
+author = option_env ! ("CARGO_PKG_AUTHORS").unwrap_or(""))]
+struct Cli {
+    /// Optional Connection Network Type
+    #[clap(long, default_value = "clearnet", possible_values = &["tor", "clearnet"])]
+    network: String,
+    /// Optional DNS data directory. Default value : "~/.coinswap/maker"
+    #[clap(long, short = 'd')]
+    data_directory: Option<PathBuf>,
+    /// Sets the full node address for rpc connection.
+    #[clap(
+        name = "ADDRESS:PORT",
+        long,
+        short = 'r',
+        default_value = "127.0.0.1:18443"
+    )]
+    pub rpc: String,
+    /// Sets the rpc basic authentication.
+    #[clap(
+        name = "USER:PASSWD",
+        short = 'a',
+        long,
+        value_parser = parse_proxy_auth,
+        default_value = "user:password",
+    )]
+    pub auth: (String, String),
```

Comment:
The authentication details should come from the env rather than being passed as a command line argument.